### ISSUE TITLE
Not able to execute simple query with EF Core 3.1.12

### ISSUE THREAD

**Issue by v-zubritsky** [linq2db/linq2db.EntityFrameworkCore#121]
Hi! I might be doing sth wrong, but spent some time and was not able to find the root cause.
Trying to use `3.9.6.0` version of your library on top of `EF Core 3.1.12`, but with no luck.

Invocation, which seems to be pretty basic, and required to just check that library works

```
dbContext.SomeDbSet.Take(20).ToLinqToDB().ToArray();
```

Fails with error
```
Method 'ExecuteAsyncEnumerable' in type 'LinqToDB.EntityFrameworkCore.Internal.LinqToDBForEFQueryProvider`1' from assembly 'linq2db.EntityFrameworkCore, Version=3.9.6.0, Culture=neutral, PublicKeyToken=e41013125f9e410a' does not have an implementation.
```

`LinqToDBForEFTools.Initialize()` is invoked prior to `DbContext` instance creation.

Will be glad to get help here. And sorry If I somehow missed some crucial part of the documentation.

**Comment by sdanyliv** [comment]
It is working with 3.1.11?

**Comment by v-zubritsky** [comment]
@sdanyliv haven't checked. Could verify, if you believe that might help.

**Comment by v-zubritsky** [comment]
@sdanyliv Here is a small repro for you. Important note, which could be actually a reason of the error, is that my app is targeting net48. 

[Repro.zip](https://github.com/linq2db/linq2db.EntityFrameworkCore/files/6116432/Repro.zip)

**Comment by v-zubritsky** [comment]
Same error on `3.9.5` and `3.9.0`, while a different one for `3.4.0`

```
System.MissingMethodException: 'Method not found: 'System.Collections.Generic.IAsyncEnumerable`1<!!0> LinqToDB.Async.IQueryProviderAsync.ExecuteAsync(System.Linq.Expressions.Expression)'.'
```

**Comment by sdanyliv** [comment]
Reproduced. Working on fix. Thanks for the reproducing sample.

**Comment by v-zubritsky** [comment]
No ideas what is happening here tbh. 

I built 3.9.6 locally and referenced it – fails on attempt to load `LinqToDBForEFQueryProvider` type blaming the same `ExecuteAsyncEnumerable` method, which looks fine to me. It's enough to just use the type anywhere in the method directly and it will throw (I guess it's happening during JIT compilation as static ctor initializing `_executeAsyncMethodInfo` seems to be irrelevant, commented it out and still got the same error).

Replicated logic of the `LinqToDBForEFTools.ToLinqToDb` without using `LinqToDBForEFQueryProvider` and it works as expected, loads data from sql server in a correct way.

**Comment by MaceWindu** [comment]
needs retesting

**Comment by v-zubritsky** [comment]
@MaceWindu Is linq2db next version release planned already?
I'm ok with using custom build of it for some time to have EF Core integration working, but it'd be nice to get it delivered from nuget.

**Comment by MaceWindu** [comment]
We plan to release it next thursday

**Comment by sdanyliv** [comment]
Fixed in 3.10.0

**Comment by v-zubritsky** [comment]
Works for me now. Confirm.
